lib/open_food_network/reports/line_items.rb

```diff
@@ -23,7 +29,7 @@ def self.list(permissions, params)
           line_item.order.ship_address.andand.assign_attributes(firstname: I18n.t('admin.reports.hidden'), lastname: "", phone: "", address1: "", address2: "", city: "", zipcode: "", state: nil)
           line_item.order.assign_attributes(email: I18n.t('admin.reports.hidden'))
         end
-        line_items
+        @list = line_items
```

Comment:
:warning:

This caches the line items, which is based on the `permissions` argument, in the class. So if we pass a `Permissions` object for a different user in the same thread, the result for the first one would be used.
   
   ```ruby
class MyClass
  def self.my_method
    return @result unless @result.nil?
    @result = Time.now
  end
end

MyClass.my_method
# => 2019-09-26 13:09:07 +0800
sleep 3
MyClass.my_method
# => 2019-09-26 13:09:07 +0800
```

Unicorn would also retain this result in memory throughout the life of the worker, which would be a memory leak.

Reply:
Oops! Yes, I just saw the build.